ecl/hqlcpp/hqlres.cpp

```diff
@@ -40,7 +41,7 @@ class ResourceItem : public CInterface
 };
 
 
-ResourceManager::ResourceManager()
+ResourceManager::ResourceManager(IHqlCppInstance &_cppInstance) : cppInstance(_cppInstance)
```

Comment:
I'll read some more, but hesitant about adding dependencies from child to parent.

Reply:
I think I needed it for adding sources. It's a bit messy as lists of sources (and flags) are maintained in quite a few places, but I was keeping the existing structure for that.

I can probably pass an interface down to the relevant calls instead if you think that is cleaner, but I don't think it's ever going to be especially clean without some major refactoring.